ambry-frontend/src/test/java/com/github/ambry/frontend/FrontendQuotaIntegrationTest.java

```diff
@@ -0,0 +1,508 @@
+/**
+ * Copyright 2021 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.frontend;
+
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.account.InMemAccountService;
+import com.github.ambry.account.InMemAccountServiceFactory;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.clustermap.Partition;
+import com.github.ambry.clustermap.PartitionState;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.commons.CommonTestUtils;
+import com.github.ambry.commons.LoggingNotificationSystem;
+import com.github.ambry.commons.NettySslFactory;
+import com.github.ambry.commons.SSLFactory;
+import com.github.ambry.commons.TestSSLUtils;
+import com.github.ambry.config.FrontendConfig;
+import com.github.ambry.config.NettyConfig;
+import com.github.ambry.config.QuotaConfig;
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.quota.QuotaMode;
+import com.github.ambry.rest.NettyClient;
+import com.github.ambry.rest.RestServer;
+import com.github.ambry.rest.RestServiceException;
+import com.github.ambry.rest.RestUtils;
+import com.github.ambry.router.ByteRange;
+import com.github.ambry.utils.TestUtils;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Integration tests for Ambry frontend with quota.
+ */
+@RunWith(Parameterized.class)
+public class FrontendQuotaIntegrationTest extends FrontendIntegrationTestBase {
```

Comment:
Btw, what is the difficulty in adding quota tests to original int-test?

Reply:
The main issue was that some of the verification methods for various requests (get/post etc) need to be different when request is actually rejected. Doing this in the same class would lead to a lot of if/else making the code hard to read.